src/lib/Store.ts

```diff
@@ -0,0 +1,108 @@
+export interface StoreConstants {
+    DB_VERSION: number;
+    DB_NAME: string;
+    DB_STORE_NAME: string;
+}
+
+export abstract class Store<Info, InfoEntry> {
+    private static INDEXEDDB_IMPLEMENTATION = window.indexedDB;
+
+    private _dbPromise: Promise<IDBDatabase> | null;
+    private _indexedDB: IDBFactory;
+
+    protected constructor() {
+        this._dbPromise = null;
+        this._indexedDB = Store.INDEXEDDB_IMPLEMENTATION;
+    }
+
+    public async get(id: string): Promise<Info | null> {
+        const db = await this.connect();
+        const transaction = db.transaction(this.getConstants().DB_STORE_NAME);
+        const request = transaction.objectStore(this.getConstants().DB_STORE_NAME).get(id);
+        const result = await this._requestAsPromise(request, transaction);
+        return result ? this.entryToInfo(result) : result;
```

Comment:
The Nimiq Core lib has many methods that are named `fromX()` and `toX()` to convert between formats, which I like. So I can imagine the `entryToInfo()` and `infoToEntry()` methods to be called `toDatabaseEntry()` and `fromDatabaseEntry()` instead (or just `toEntry()` and `fromEntry()`). That would also remove the need to call everything that goes in and out of the database an `*Info` object.

Reply:
Both functions on the Store are protected and used exclusively internally. It mostly comes from the generic being named `Store<Info, InfoEntry>`. In case of `CashlinkStore` the actual used function are `Cashlink.toObject()` and `Cashlink.fromObject()` respectively. Those also are publicly exposed.

I agree to renaming the two abstract functions to `fromEntry` and `toEntry`, as it is more consistent with the naming we already established (and also because I like it better as well). For the generic I am open to suggestions to especially the `Info` part of it.